ssl/statem/statem_srvr.c

```diff
@@ -1781,6 +1781,16 @@ int tls_construct_server_key_exchange(SSL *s)
             n += 1 + nr[i];
         else
 #endif
+#ifndef OPENSSL_NO_DH
+        /*
+         * for interoperability with some versions of the Microsoft TLS
+         * stack, we need to zero pad the DHE pub key to the same length
+         * as the prime, so include the length of the prime here
+         */
+        if ((i == 2) && (type & (SSL_kDHE | SSL_kDHEPSK)) && (nr[0] > nr[2]))
```

Comment:
That expression is not strictly required, I felt it was a bit clearer, but if nr[0] == nr[2], the inside of the if still works (for both clauses), and if nr[0] < nr[2], something terrible has happened elsewhere.